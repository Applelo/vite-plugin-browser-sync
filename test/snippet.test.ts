import { resolve } from 'path'
import fetch from 'node-fetch'
import { it, expect } from 'vitest'
import { createServer } from 'vite'
import VitePluginBrowserSync from '../src'

it('snippet option', async () => {
  const server = await createServer({
    // any valid user config options, plus `mode` and `configFile`
    configFile: false,
    root: resolve(__dirname, './../demo'),
    plugins: [VitePluginBrowserSync({ mode: 'snippet' })]
  })
  await server.listen()

  const text = await fetch('http://127.0.0.1:5173/').then(res => res.text())

  server.close()
  expect(text).matchSnapshot()
})
