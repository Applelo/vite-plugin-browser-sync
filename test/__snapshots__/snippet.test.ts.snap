// Vitest Snapshot v1

exports[`snippet option 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <script type=\\"module\\" src=\\"/@vite/client\\"></script>

    <meta charset=\\"UTF-8\\" />
    <link rel=\\"icon\\" type=\\"image/svg+xml\\" href=\\"/vite.svg\\" />
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\" />
    <title>Vite Plugin Browser Sync</title>
  </head>
  <body>
    <h1>Vite Plugin Browser Sync</h1>

    <form id=\\"form\\">
      <div>
        <label for=\\"firstname\\">First Name</label>
        <input type=\\"text\\" id=\\"firstname\\" name=\\"firstname\\" />
      </div>
      <div>
        <label for=\\"lastname\\">Last Name</label>
        <input type=\\"text\\" id=\\"lastname\\" name=\\"lastname\\" />
      </div>
      <input type=\\"submit\\" value=\\"Submit\\" />
    </form>
    <p id=\\"message\\"></p>

    <script type=\\"module\\" src=\\"/src/main.ts\\"></script>
    <script async=\\"\\" src=\\"http://localhost:3000/browser-sync/browser-sync-client.js?v=2.27.10\\"></script>
  </body>
</html>
"
`;
